### PR TITLE
Projectile editor

### PR DESCRIPTION
**Pull request by BlackRece**
**Describe your changes**
Fully functioning Projectile editor that saves and loads to json file and when clicking "Test Fire" will show how the projectiles will look.

**Screenshots**
image

### CODE REVIEW

File: Studio Sim/Objects/Projectile.h [hunk 1]
```diff
@@ -8,29 +8,50 @@ class Graphics;
...
+	void SpawnProjectile(Vector2f vSpawnPosition, float fLifeTime);
 	
 private:
+	inline Vector2f CalcWaveVector(const float fPeak, const float fCycles, const float fDelta) noexcept {
+		return Vector2f(
+			fPeak * sinf(AI_MATH_TWO_PI_F * fCycles * fDelta),
+			fPeak * cosf(AI_MATH_TWO_PI_F * fCycles * fDelta)
+		);
+	}
+
```
Comment (juliusz-kaczmarek) [inline Studio Sim/Objects/Projectile.h:40]: This might be incorrect

File: Studio Sim/Projectile Editor/ProjectileEditor.cpp [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+					msg = "Traveling Angle Adjustment: " + std::to_string(m_vecManagers[iManIndex].m_vecProjectiles[iProIndex].m_fWaveAngle);
+					ImGui::DragFloat(
+						std::string("Wave##Man")
+							.append(std::to_string(iManIndex))
+							.append("Pro")
+							.append(std::to_string(iProIndex))
+							.c_str(),
+						&m_vecManagers[iManIndex].m_vecProjectiles[iProIndex].m_fWaveAngle,
+						0.1f, -360.0f, 360.0f, "%.2f");
```
Comment (juliusz-kaczmarek) [inline Studio Sim/Projectile Editor/ProjectileEditor.cpp:212]: This might be incorrect

**Review by juliusz-kaczmarek — CHANGES_REQUESTED** [review]
The wiggle doesn't seem to happen, it's more like going in some weird pattern, not sure why this is happening, but I have a suspicion the sine wave formula is applied incorrectly

File: Studio Sim/Objects/Projectile.h (hunk 1 above)
Comment (BlackRece) [reply Studio Sim/Objects/Projectile.h:40]: I think the effect of this formula is correct but it's the wrong formula.
Please suggest what formula you wish for me to use/simulate.

File: Studio Sim/Projectile Editor/ProjectileEditor.cpp (hunk 2 above)
Comment (BlackRece) [reply Studio Sim/Projectile Editor/ProjectileEditor.cpp:210]: image
Currently only displays as "Wave" have corrected and will commit.

Comment (BlackRece) [reply Studio Sim/Projectile Editor/ProjectileEditor.cpp:210]: Will also change depending on the requirements of a team-agreed wave function.

Comment (juliusz-kaczmarek) [reply Studio Sim/Projectile Editor/ProjectileEditor.cpp]: image
Expected fly pattern with basic sine wave projectile

**Review by kyle-robinson — CHANGES_REQUESTED** [review]
Other than the changes that Juliusz requested, I think I'm happy enough.

I'll mark this as a change request though, since after trying to resolve the conflicts, it doesn't seem like the editor is working anymore. Or at the very least, it doesn't look like the projectiles are doing anything, and when trying to modify any of the sliders after loading a pattern causes a crash.

Might need to go through this together at some point tomorrow. I won't push the changes yet since it is technically broken on my end at the minute 😏

**Review by kyle-robinson — APPROVED** [review]

**Review by juliusz-kaczmarek — APPROVED** [review]

**Pull request merged**